rust/src/lib.rs

```diff
@@ -36,6 +38,33 @@ mod ffi {
     extern "Rust" {
         fn download_to_fd(url: &str) -> Result<i32>;
     }
+
+    #[derive(Default)]
+    /// A copy of LiveFsState that is bridged to C++; the main
+    /// change here is we can't use Option<> yet, so empty values
+    /// are represented by the empty string.
+    struct LiveApplyState {
```

Comment:
Is it worth maintaining two separate types for this instead of just converging on the cxxbridge-compatible one? At least Rust has `String::is_empty()` which isn't too bad.

Reply:
OK I did a rework to more clearly demote the `Option<>`-using version to just for serialization.  I already made a logic error where I was missing a `!` before `is_empty()` in a `From<>` implementation but eh it was missing unit tests anyways.

At some point either cxx-rs will support `Option<>` or more of this logic will move to Rust and we'll be able to make `Option<>` usage more prominent again.  So all of this is about some shorter term ugliness/tech debt in some areas in return for much larger improvements in others.